From 4f3c288a1bac741fbcc9a8cd790d30618093a03e Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sun, 18 Jan 2015 07:38:58 +0000
Subject: [PATCH] lua: Define luaL_checkint which was deprecated in lua 5.3.

---
 generator/lua.ml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/generator/lua.ml b/generator/lua.ml
index c1fa6f0..f2aa8c7 100644
--- a/generator/lua.ml
+++ b/generator/lua.ml
@@ -54,6 +54,12 @@ let generate_lua_c () =
 #endif
 #endif
 
+#if LUA_VERSION_NUM >= 503
+#ifndef luaL_checkint
+#define luaL_checkint(L,n) ((int)luaL_checkinteger(L, (n)))
+#endif
+#endif
+
 #include <guestfs.h>
 #include \"guestfs-internal-frontend.h\"
 
-- 
2.2.1

