From 5706aaf0542fb67ba9398076f2531911bdd95dd3 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Sat, 9 Mar 2013 18:36:09 +0000
Subject: [PATCH 01/36] gnulib: Use module mkstemps for RHEL 5 glibc which lacked this function.

---
 bootstrap     |    1 +
 m4/.gitignore |    1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/bootstrap b/bootstrap
index be0dfe4..931646b 100755
--- a/bootstrap
+++ b/bootstrap
@@ -68,6 +68,7 @@ maintainer-makefile
 manywarnings
 memmem
 mkdtemp
+mkstemps
 netdb
 netinet_in
 openat
diff --git a/m4/.gitignore b/m4/.gitignore
index 0d903b7..b827552 100644
--- a/m4/.gitignore
+++ b/m4/.gitignore
@@ -138,6 +138,7 @@
 /mgetgroups.m4
 /mkdir.m4
 /mkdtemp.m4
+/mkstemps.m4
 /mmap-anon.m4
 /mode_t.m4
 /msvc-inval.m4
-- 
1.7.4.1

