From 4f868b84d6e7c387e27f50aae54476bc97a70bc2 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sun, 22 Jan 2012 14:38:07 +0000
Subject: [PATCH 1/2] php: Enable 'make clean' in extension subdirectory.

---
 php/Makefile.am |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/php/Makefile.am b/php/Makefile.am
index 828960c..7652b66 100644
--- a/php/Makefile.am
+++ b/php/Makefile.am
@@ -49,10 +49,10 @@ extension/config.h: extension/config.m4 ../config.status
 
 TESTS = run-php-tests.sh
 
-# Amazingly the PHP-installed extension Makefile doesn't have a
-# 'make clean' rule.
-#clean-local:
-#	$(MAKE) -C extension clean
+# Before PHP 5.4.0, the extension directory didn't have a 'make clean'
+# rule.  Therefore, don't die if this fails.
+clean-local:
+	-$(MAKE) -C extension clean
 
 install-data-hook:
 	$(MAKE) -C extension INSTALL_ROOT=$(DESTDIR) install
-- 
1.7.6

