From 64498034bbc7bce2692597fb8bfbb721e8180543 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 25 Jul 2014 10:04:44 +0100
Subject: [PATCH] tests: lvm: Allow test to be skipped.

---
 tests/lvm/test-lvm-filtering.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tests/lvm/test-lvm-filtering.sh b/tests/lvm/test-lvm-filtering.sh
index a448554..3744021 100755
--- a/tests/lvm/test-lvm-filtering.sh
+++ b/tests/lvm/test-lvm-filtering.sh
@@ -20,6 +20,11 @@
 
 set -e
 
+if [ -n "$SKIP_TEST_LVM_FILTERING_SH" ]; then
+    echo "$0: skipping test because environment variable is set."
+    exit 77
+fi
+
 rm -f test-lvm-filtering-1.img test-lvm-filtering-2.img
 
 actual=$(../../fish/guestfish <<'EOF'
-- 
1.9.0

