--- libguestfs-1.19.42.old/src/launch-libvirt.c	2012-09-18 08:27:39.242142221 +0100
+++ libguestfs-1.19.42/src/launch-libvirt.c	2012-09-18 08:35:46.604801952 +0100
@@ -115,7 +115,7 @@
   int console = -1, r;
   uint32_t size;
   void *buf = NULL;
-  int disable_svirt = is_custom_qemu (g);
+  const int disable_svirt = 1;
 
   /* At present you must add drives before starting the appliance.  In
    * future when we enable hotplugging you won't need to do this.
