From 5ab3a75e1dba7cacd45ff9a094bf27c517edd117 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 17 Jan 2015 14:24:35 +0000
Subject: [PATCH] tests/lvm: Allow test-lvm-mapping.pl to be skipped.

---
 tests/lvm/test-lvm-mapping.pl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/lvm/test-lvm-mapping.pl b/tests/lvm/test-lvm-mapping.pl
index 79a5bdd..2509231 100755
--- a/tests/lvm/test-lvm-mapping.pl
+++ b/tests/lvm/test-lvm-mapping.pl
@@ -22,6 +22,8 @@ use warnings;
 
 use Sys::Guestfs;
 
+exit 77 if $ENV{SKIP_TEST_LVM_MAPPING_PL};
+
 my $g = Sys::Guestfs->new ();
 
 $g->add_drive_scratch (256 * 1024 * 1024);
-- 
2.1.0

