From 1668c3fcc9a67e59c98121a473d68087d105d334 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Wed, 2 Feb 2011 20:26:19 +0000
Subject: [PATCH 2/2] regressions: Fix rhbz557655.sh so it works with tracing enabled.

---
 regressions/rhbz557655.sh |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/regressions/rhbz557655.sh b/regressions/rhbz557655.sh
index 2306147..2bc10cd 100755
--- a/regressions/rhbz557655.sh
+++ b/regressions/rhbz557655.sh
@@ -77,6 +77,7 @@ EOF
 # number of truncate_size.
 mv test.err test.err~
 grep -E 'set[-_]memsize|truncate[-_]size' test.err~ |
+  grep -Ev 'libguestfs: trace:' |
   grep -Ev 'proc 200' > test.err
 rm test.err~
 
-- 
1.7.3.5

