From 66035f3e05e6cd9a87f8b59605d38bc5cd7b97fa Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Sat, 16 Jun 2012 10:46:20 +0100
Subject: [PATCH 03/33] EPEL 5: Replace macro AC_STRUCT_DIRENT_D_TYPE for ancient autoconf.

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index a0d8007..81f5466 100644
--- a/configure.ac
+++ b/configure.ac
@@ -258,7 +258,7 @@ The code will still compile, but is likely to leak memory and other
 resources when it runs.])])
 
 dnl Check if dirent (readdir) supports d_type member.
-AC_STRUCT_DIRENT_D_TYPE
+AC_DEFINE([HAVE_STRUCT_DIRENT_D_TYPE],[1],[Define to 1 if 'struct dirent' has member 'd_type'])
 
 dnl Check if stat has the required fields.
 AC_STRUCT_ST_BLOCKS
-- 
1.7.4.1

