From 03263b3a4fe4abce8dcdd53fb1880eca64ef46ef Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Sat, 16 Jun 2012 10:46:20 +0100
Subject: [PATCH 07/37] EPEL 5: Replace macro AC_STRUCT_DIRENT_D_TYPE for ancient autoconf.

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 38998ee..ad2897d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -251,7 +251,7 @@ The code will still compile, but is likely to leak memory and other
 resources when it runs.])])
 
 dnl Check if dirent (readdir) supports d_type member.
-AC_STRUCT_DIRENT_D_TYPE
+AC_DEFINE([HAVE_STRUCT_DIRENT_D_TYPE],[1],[Define to 1 if 'struct dirent' has member 'd_type'])
 
 dnl Check if stat has the required fields.
 AC_STRUCT_ST_BLOCKS
-- 
1.7.4.1

