From efd8d11ac4e89909ea207d5fc1c5da3ecfee9ec3 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Mon, 18 Jun 2012 08:35:14 +0100
Subject: [PATCH 11/19] EPEL 5: Replace format6 with format4 in OCaml gettext routines.

---
 configure.ac |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index d18c76b..87e6723 100644
--- a/configure.ac
+++ b/configure.ac
@@ -786,14 +786,14 @@ EOF
             cat <<EOF >>$output
 module Gettext = struct
   external s_ : string -> string = "%identity"
-  external f_ : ('a, 'b, 'c, 'd, 'e, 'f) format6
-    -> ('a, 'b, 'c, 'd, 'e, 'f) format6
+  external f_ : ('a, 'b, 'c, 'd) format4
+    -> ('a, 'b, 'c, 'd) format4
     = "%identity"
   let sn_ : string -> string -> int -> string
     = fun s p n -> if n = 1 then s else p
-  let fn_ : ('a, 'b, 'c, 'd, 'e, 'f) format6
-    -> ('a, 'b, 'c, 'd, 'e, 'f) format6
-    -> int -> ('a, 'b, 'c, 'd, 'e, 'f) format6
+  let fn_ : ('a, 'b, 'c, 'd) format4
+    -> ('a, 'b, 'c, 'd) format4
+    -> int -> ('a, 'b, 'c, 'd) format4
     = fun s p n -> if n = 1 then s else p
 end
 EOF
-- 
1.7.4.1

