From dc0acbad966bf4bfa4e211957707ffe300a805d3 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Mon, 18 Jun 2012 17:21:30 +0100
Subject: [PATCH 13/20] EPEL 5: Remove id= from -drive parameter on qemu command line.

---
 src/launch.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/launch.c b/src/launch.c
index 3bb41ae..2037a67 100644
--- a/src/launch.c
+++ b/src/launch.c
@@ -722,7 +722,7 @@ launch_appliance (guestfs_h *g)
 
       char buf2[PATH_MAX + 64];
       add_cmdline (g, "-drive");
-      snprintf (buf2, sizeof buf2, "file=%s,snapshot=on,id=appliance,if=%s%s",
+      snprintf (buf2, sizeof buf2, "file=%s,snapshot=on,if=%s%s",
                 appliance, virtio_scsi ? "none" : "virtio", cachemode);
       add_cmdline (g, buf2);
 
@@ -1568,12 +1568,11 @@ qemu_drive_param (guestfs_h *g, const struct drive *drv, size_t index)
   else
     iface = "virtio";
 
-  snprintf (&r[i], len-i, "%s%s%s%s,id=hd%zu,if=%s",
+  snprintf (&r[i], len-i, "%s%s%s%s,if=%s",
             drv->readonly ? ",snapshot=on" : "",
             drv->use_cache_off ? ",cache=off" : "",
             drv->format ? ",format=" : "",
             drv->format ? drv->format : "",
-            index,
             iface);
 
   return r;                     /* caller frees */
-- 
1.7.4.1

