From dd6987286760d8423b8cc7f5fba5b9720adc85a2 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Mon, 18 Jun 2012 17:21:30 +0100
Subject: [PATCH 17/38] EPEL 5: Remove id= from -drive parameter on qemu command line.

---
 src/launch-appliance.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/launch-appliance.c b/src/launch-appliance.c
index 10f63f1..62ab76e 100644
--- a/src/launch-appliance.c
+++ b/src/launch-appliance.c
@@ -339,7 +339,7 @@ launch_appliance (guestfs_h *g, const char *arg)
       size_t buf2_len = strlen (appliance) + 64;
       char buf2[buf2_len];
       add_cmdline (g, "-drive");
-      snprintf (buf2, buf2_len, "file=%s,snapshot=on,id=appliance,if=%s%s",
+      snprintf (buf2, buf2_len, "file=%s,snapshot=on,if=%s%s",
                 appliance, virtio_scsi ? "none" : "virtio", cachemode);
       add_cmdline (g, buf2);
 
@@ -957,14 +957,13 @@ qemu_drive_param (guestfs_h *g, const struct drive *drv, size_t index)
   else
     iface = "virtio";
 
-  snprintf (&r[i], len-i, "%s%s%s%s%s%s,id=hd%zu,if=%s",
+  snprintf (&r[i], len-i, "%s%s%s%s%s%s,if=%s",
             drv->readonly ? ",snapshot=on" : "",
             drv->use_cache_none ? ",cache=none" : "",
             drv->format ? ",format=" : "",
             drv->format ? drv->format : "",
             drv->disk_label ? ",serial=" : "",
             drv->disk_label ? drv->disk_label : "",
-            index,
             iface);
 
   return r;                     /* caller frees */
-- 
1.7.4.1

