From b3995d6a0c981983dbb311a698f2c044e286406b Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Thu, 30 Aug 2012 17:57:46 +0100
Subject: [PATCH 17/29] EPEL 5: Disable libvirt attach method.

---
 src/launch-libvirt.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/launch-libvirt.c b/src/launch-libvirt.c
index 9f14add..96e72d2 100644
--- a/src/launch-libvirt.c
+++ b/src/launch-libvirt.c
@@ -66,7 +66,7 @@
                              MIN_LIBVIRT_MINOR * 1000 + \
                              MIN_LIBVIRT_MICRO)
 
-#if defined(HAVE_LIBVIRT) && \
+#if 0 && defined(HAVE_LIBVIRT) && \
   LIBVIR_VERSION_NUMBER >= MIN_LIBVIRT_VERSION && \
   defined(HAVE_LIBXML2)
 
-- 
1.7.4.1

