From 4285778047a444c8f5fb09c09d4e5ea59a7c0c33 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Thu, 30 Aug 2012 17:57:46 +0100
Subject: [PATCH 18/36] EPEL 5: Disable libvirt attach method.

---
 src/launch-libvirt.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/launch-libvirt.c b/src/launch-libvirt.c
index e549ae8..c013dee 100644
--- a/src/launch-libvirt.c
+++ b/src/launch-libvirt.c
@@ -72,7 +72,7 @@
                              MIN_LIBVIRT_MINOR * 1000 + \
                              MIN_LIBVIRT_MICRO)
 
-#if defined(HAVE_LIBVIRT) && \
+#if 0 && defined(HAVE_LIBVIRT) && \
   LIBVIR_VERSION_NUMBER >= MIN_LIBVIRT_VERSION && \
   defined(HAVE_LIBXML2)
 
-- 
1.7.4.1

