From 73cf29101f55e766c781dc241ae6e33228070cf9 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 8 Mar 2014 17:56:43 +0000
Subject: [PATCH] Update generated files.

---
 gobject/Makefile.inc | 2 ++
 po/POTFILES          | 1 +
 2 files changed, 3 insertions(+)

diff --git a/gobject/Makefile.inc b/gobject/Makefile.inc
index 4f2927e..8a068d8 100644
--- a/gobject/Makefile.inc
+++ b/gobject/Makefile.inc
@@ -50,6 +50,7 @@ guestfs_gobject_headers= \
   include/guestfs-gobject/optargs-btrfs_fsck.h \
   include/guestfs-gobject/optargs-compress_device_out.h \
   include/guestfs-gobject/optargs-compress_out.h \
+  include/guestfs-gobject/optargs-copy_attributes.h \
   include/guestfs-gobject/optargs-copy_device_to_device.h \
   include/guestfs-gobject/optargs-copy_device_to_file.h \
   include/guestfs-gobject/optargs-copy_file_to_device.h \
@@ -124,6 +125,7 @@ guestfs_gobject_sources= \
   src/optargs-btrfs_fsck.c \
   src/optargs-compress_device_out.c \
   src/optargs-compress_out.c \
+  src/optargs-copy_attributes.c \
   src/optargs-copy_device_to_device.c \
   src/optargs-copy_device_to_file.c \
   src/optargs-copy_file_to_device.c \
diff --git a/po/POTFILES b/po/POTFILES
index 090871b..fd35411 100644
--- a/po/POTFILES
+++ b/po/POTFILES
@@ -164,6 +164,7 @@ gobject/src/optargs-btrfs_filesystem_resize.c
 gobject/src/optargs-btrfs_fsck.c
 gobject/src/optargs-compress_device_out.c
 gobject/src/optargs-compress_out.c
+gobject/src/optargs-copy_attributes.c
 gobject/src/optargs-copy_device_to_device.c
 gobject/src/optargs-copy_device_to_file.c
 gobject/src/optargs-copy_file_to_device.c
-- 
1.8.5.3

