--- libguestfs-1.21.23.old/generator/actions.ml	2013-03-19 12:24:40.000000000 +0000
+++ libguestfs-1.21.23/generator/actions.ml	2013-03-28 23:58:07.738134412 +0000
@@ -3681,7 +3681,7 @@
     proc_nr = Some 49;
     tests = [
       InitISOFS, Always, TestOutput (
-        [["file"; "/empty"]], "empty");
+        [["file"; "/empty"]], "empty ");
       InitISOFS, Always, TestOutput (
         [["file"; "/known-1"]], "ASCII text");
       InitISOFS, Always, TestLastFail (
