diff -ur libguestfs-1.5.16.orig/regressions/Makefile.am libguestfs-1.5.16/regressions/Makefile.am
--- libguestfs-1.5.16.orig/regressions/Makefile.am	2010-09-10 13:13:56.000000000 +0100
+++ libguestfs-1.5.16/regressions/Makefile.am	2010-09-16 11:52:56.042767630 +0100
@@ -34,7 +34,6 @@
 	rhbz580246.sh \
 	test-cancellation-download-librarycancels.sh \
 	test-cancellation-upload-daemoncancels.sh \
-	test-copy.sh \
 	test-find0.sh \
 	test-luks.sh \
 	test-lvm-filtering.sh \
@@ -49,7 +48,8 @@
 	test-stringlist.sh
 
 SKIPPED_TESTS = \
-	test-bootbootboot.sh
+	test-bootbootboot.sh \
+	test-copy.sh
 
 FAILING_TESTS = \
 	test-qemudie-launchfail.sh
diff -ur libguestfs-1.5.16.orig/regressions/Makefile.in libguestfs-1.5.16/regressions/Makefile.in
--- libguestfs-1.5.16.orig/regressions/Makefile.in	2010-09-15 22:09:52.000000000 +0100
+++ libguestfs-1.5.16/regressions/Makefile.in	2010-09-16 11:57:14.320115536 +0100
@@ -79,12 +79,12 @@
 	rhbz503169c10.sh rhbz503169c13.sh rhbz557655.sh rhbz576879.sh \
 	rhbz578407.sh rhbz580246.sh \
 	test-cancellation-download-librarycancels.sh \
-	test-cancellation-upload-daemoncancels.sh test-copy.sh \
-	test-find0.sh test-luks.sh test-lvm-filtering.sh \
-	test-lvm-mapping.pl test-noexec-stack.pl \
-	test-qemudie-killsub.sh test-qemudie-midcommand.sh \
-	test-qemudie-synch.sh test-read_file.sh test-remote.sh \
-	test-reopen.sh test-stringlist.sh
+	test-cancellation-upload-daemoncancels.sh test-find0.sh \
+	test-luks.sh test-lvm-filtering.sh test-lvm-mapping.pl \
+	test-noexec-stack.pl test-qemudie-killsub.sh \
+	test-qemudie-midcommand.sh test-qemudie-synch.sh \
+	test-read_file.sh test-remote.sh test-reopen.sh \
+	test-stringlist.sh
 check_PROGRAMS = rhbz501893$(EXEEXT)
 subdir = regressions
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -987,7 +987,8 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SKIPPED_TESTS = \
-	test-bootbootboot.sh
+	test-bootbootboot.sh \
+	test-copy.sh
 
 FAILING_TESTS = \
 	test-qemudie-launchfail.sh
