Only in libguestfs-1.12.0: autom4te.cache
diff -ur libguestfs-1.12.0.old/configure libguestfs-1.12.0/configure
--- libguestfs-1.12.0.old/configure	2011-07-20 16:40:27.000000000 +0100
+++ libguestfs-1.12.0/configure	2011-07-20 18:48:50.261315374 +0100
@@ -36849,7 +36849,7 @@
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for virtio-serial support in $QEMU" >&5
 $as_echo_n "checking for virtio-serial support in $QEMU... " >&6; }
-    if $QEMU -nographic -device \? 2>&1 | grep -sq virtio-serial; then
+    if $QEMU -machine pc,accel=tcg -nographic -device \? 2>&1 | grep -sq virtio-serial; then
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
     else
diff -ur libguestfs-1.12.0.old/configure.ac libguestfs-1.12.0/configure.ac
--- libguestfs-1.12.0.old/configure.ac	2011-07-20 16:35:50.000000000 +0100
+++ libguestfs-1.12.0/configure.ac	2011-07-20 18:48:41.251139811 +0100
@@ -326,7 +326,7 @@
     fi
 
     AC_MSG_CHECKING([for virtio-serial support in $QEMU])
-    if $QEMU -nographic -device \? 2>&1 | grep -sq virtio-serial; then
+    if $QEMU -machine pc,accel=tcg -nographic -device \? 2>&1 | grep -sq virtio-serial; then
         AC_MSG_RESULT([yes])
     else
         AC_MSG_RESULT([no])
