diff -ur libguestfs-1.7.13.old/fuse/Makefile.am libguestfs-1.7.13/fuse/Makefile.am
--- libguestfs-1.7.13.old/fuse/Makefile.am	2010-11-24 20:12:08.000000000 +0000
+++ libguestfs-1.7.13/fuse/Makefile.am	2010-11-24 22:31:13.522116969 +0000
@@ -77,8 +77,8 @@
 
 # Tests.
 
-TESTS = test-fuse.sh
-TESTS_ENVIRONMENT = \
-	top_builddir=..
+#TESTS = test-fuse.sh
+#TESTS_ENVIRONMENT = \
+#	top_builddir=..
 
 endif
Only in libguestfs-1.7.13/fuse: Makefile.am.orig
diff -ur libguestfs-1.7.13.old/fuse/Makefile.in libguestfs-1.7.13/fuse/Makefile.in
--- libguestfs-1.7.13.old/fuse/Makefile.in	2010-11-24 20:17:24.000000000 +0000
+++ libguestfs-1.7.13/fuse/Makefile.in	2010-11-24 22:31:13.524116997 +0000
@@ -1055,9 +1055,9 @@
 @HAVE_FUSE_TRUE@noinst_DATA = $(top_builddir)/html/guestmount.1.html
 
 # Tests.
-@HAVE_FUSE_TRUE@TESTS = test-fuse.sh
-@HAVE_FUSE_TRUE@TESTS_ENVIRONMENT = \
-@HAVE_FUSE_TRUE@	top_builddir=..
+#@HAVE_FUSE_TRUE@TESTS = test-fuse.sh
+#@HAVE_FUSE_TRUE@TESTS_ENVIRONMENT = \
+#@HAVE_FUSE_TRUE@	top_builddir=..
 
 all: all-am
 
Only in libguestfs-1.7.13/fuse: Makefile.in.orig
