From 61adbba900a4150110d9fcd20097b3a5c249e6c7 Mon Sep 17 00:00:00 2001
From: Richard Jones <rjones@redhat.com>
Date: Mon, 28 Jun 2010 09:13:36 +0100
Subject: [PATCH 1/2] Fix gfs2 support by adding required kernel modules.

---
 appliance/kmod.whitelist.in |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/appliance/kmod.whitelist.in b/appliance/kmod.whitelist.in
index b225f30..0a92122 100644
--- a/appliance/kmod.whitelist.in
+++ b/appliance/kmod.whitelist.in
@@ -65,3 +65,7 @@ crc*.ko
 libcrc*.ko
 
 loop.ko
+
+gfs2.ko
+dlm.ko
+configfs.ko
-- 
1.6.6.1

