From 887a79d1d1c47e537fdffd1208acbacd598b0f96 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Sat, 2 Oct 2010 09:07:42 +0100
Subject: [PATCH 1/2] test-virt-resize: Skip this test on 32 bit hosts.

---
 tools/test-virt-resize.sh |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/tools/test-virt-resize.sh b/tools/test-virt-resize.sh
index 75d5147..2c343cb 100755
--- a/tools/test-virt-resize.sh
+++ b/tools/test-virt-resize.sh
@@ -1,5 +1,12 @@
 #!/bin/bash -
 
+# virt-resize does not work on 32 bit because of limitations in Perl
+# so short-circuit this test on a 32 bit host.
+perl -e 'exit 1 if ~1 == 4294967294' || {
+    echo "$0: Skipping this test on 32 bit."
+    exit 0
+}
+
 export LANG=C
 set -e
 
-- 
1.7.3.1

