diff --git a/appliance/make.sh.in b/appliance/make.sh.in
index b3cbe92..677e995 100755
--- a/appliance/make.sh.in
+++ b/appliance/make.sh.in
@@ -98,6 +98,23 @@ if [ "@DIST@" = "REDHAT" ]; then
     /usr/sbin/tcpd \
     /lib/dbus-1/dbus-daemon-launch-helper \
     /lib64/dbus-1/dbus-daemon-launch-helper \
+    /etc/passwd- \
+    /etc/group- \
+    /etc/gshadow \
+    /etc/gshadow- \
+    /etc/shadow \
+    /etc/shadow- \
+    /etc/securetty \
+    /etc/sysconfig/iptables-config \
+    /etc/default/useradd \
+    /etc/security/opasswd \
+    /etc/libaudit.conf \
+    /var/log/tallylog \
+    /var/log/maillog \
+    /var/log/secure \
+    /var/log/spooler \
+    /var/log/messages \
+    /var/log/btmp \
     $(cd initramfs && echo usr/sbin/glibc_post_upgrade.*)
 
   # Kernel modules take up nearly half of the image.  Only include ones
