From 5df8c2d991639d08736a9a2f47e6ae0c768df390 Mon Sep 17 00:00:00 2001
From: Richard Jones <rjones@redhat.com>
Date: Fri, 27 Aug 2010 10:29:46 +0100
Subject: [PATCH] configure: Add backported features to version extra field.

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 97c797c..ce25d04 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,7 +20,7 @@ m4_define([libguestfs_major],   [1])
 m4_define([libguestfs_minor],   [4])
 m4_define([libguestfs_release], [3])
 # extra can be any string
-m4_define([libguestfs_extra],   [])
+m4_define([libguestfs_extra],   [+fastdf+luks+islv+inspection+ext2+serial+core])
 
 AC_INIT([libguestfs],libguestfs_major.libguestfs_minor.libguestfs_release)
 AC_CONFIG_AUX_DIR([build-aux])
-- 
1.7.1

