diff --git a/appliance/init b/appliance/init
index cda6453..becc396 100755
--- a/appliance/init
+++ b/appliance/init
@@ -67,7 +67,7 @@ lvm vgchange -ay --ignorelockingfailure
 for f in /sys/block/vd*/queue/rotational; do echo 1 > $f; done
 
 # http://kbase.redhat.com/faq/docs/DOC-5428
-for f in /sys/block/[hsv]d*/queue/scheduler; do echo noop > $f; done
+#for f in /sys/block/[hsv]d*/queue/scheduler; do echo noop > $f; done
 
 # These are useful when debugging.
 if grep -sq guestfs_verbose=1 /proc/cmdline; then
