diff -ur libguestfs-1.21.1.old/ruby/Makefile.am libguestfs-1.21.1/ruby/Makefile.am
--- libguestfs-1.21.1.old/ruby/Makefile.am	2012-12-15 19:42:02.000000000 +0000
+++ libguestfs-1.21.1/ruby/Makefile.am	2012-12-17 14:03:27.396861323 +0000
@@ -51,14 +51,14 @@
 	$(RAKE) build
 	$(RAKE) rdoc
 
-RUBY_SITELIB := $(shell $(RUBY) -rrbconfig -e "puts RbConfig::CONFIG['sitelibdir']")
-RUBY_SITEARCH := $(shell $(RUBY) -rrbconfig -e "puts RbConfig::CONFIG['sitearchdir']")
+RUBY_VENDORLIB := $(shell $(RUBY) -rrbconfig -e "puts RbConfig::CONFIG['vendorlibdir']")
+RUBY_VENDORARCH := $(shell $(RUBY) -rrbconfig -e "puts RbConfig::CONFIG['vendorarchdir']")
 
 install:
-	$(MKDIR_P) $(DESTDIR)$(RUBY_SITELIB)
-	$(MKDIR_P) $(DESTDIR)$(RUBY_SITEARCH)
-	$(INSTALL) -p -m 0644 $(srcdir)/lib/guestfs.rb $(DESTDIR)$(RUBY_SITELIB)
-	$(INSTALL) -p -m 0755 $(builddir)/ext/guestfs/_guestfs.so $(DESTDIR)$(RUBY_SITEARCH)
+	$(MKDIR_P) $(DESTDIR)$(RUBY_VENDORLIB)
+	$(MKDIR_P) $(DESTDIR)$(RUBY_VENDORARCH)
+	$(INSTALL) -p -m 0644 $(srcdir)/lib/guestfs.rb $(DESTDIR)$(RUBY_VENDORLIB)
+	$(INSTALL) -p -m 0755 $(builddir)/ext/guestfs/_guestfs.so $(DESTDIR)$(RUBY_VENDORARCH)
 
 TESTS = run-bindtests run-ruby-tests
 
